Modules/SubModule_Import/xASL_bids_BIDS2Legacy.m

```diff
@@ -119,133 +79,100 @@
-    % same folder layer, so we only use SubjectSession
     
+    % Subject-wise processing (because xASL_Iteration runs over subjects)
+    if ~isempty(regexpi(SubjectID,x.SUBJECT))
     
-    %% 3. Define SubjectVisit
-    iVisit = find(strcmp(BIDS.sessionName, SessionID));
-    % remove iteration for iVisit=1 % iterate visit/session in this "BIDS.subjects" (always 1 session per BIDS.subjects)
-    % ExploreASL uses visit as a number (e.g. _1 _2 _3 etc)
-    if nVisits==1
-        pathLegacy_SubjectVisit = fullfile(pathLegacy, SubjectID);
-        VisitString = '';
-    else
-        if isempty(iVisit)
-            fprintf('\nEmpty session number, setting session number to 1...\n');
-            iVisit = 1;
+        % Session ID (Currently, ExploreASL concatenates subject_visit/timepoint in the same folder layer, so we only use SubjectSession)
+        SessionID = BIDS.subjects(iSubjSess).session;
+
+
+        %% 3. Define SubjectVisit
+        iVisit = find(strcmp(BIDS.sessionName, SessionID));
+        % remove iteration for iVisit=1 % iterate visit/session in this "BIDS.subjects" (always 1 session per BIDS.subjects)
+        % ExploreASL uses visit as a number (e.g. _1 _2 _3 etc)
+        if nVisits==1
+            pathLegacy_SubjectVisit = fullfile(pathLegacy, SubjectID);
+            VisitString = '';
+        else
+            if isempty(iVisit)
+                fprintf('\nEmpty session number, setting session number to 1...\n');
+                iVisit = 1;
+            end
+
+            pathLegacy_SubjectVisit = fullfile(pathLegacy, [SubjectID '_' xASL_num2str(iVisit)]);
+            VisitString = [' visit ' SessionID];
         end
-        
-        pathLegacy_SubjectVisit = fullfile(pathLegacy, [SubjectID '_' xASL_num2str(iVisit)]);
-        VisitString = [' visit ' SessionID];
+        SubjectVisit = [SubjectID VisitString];
+        xASL_adm_CreateDir(pathLegacy_SubjectVisit);
+
+
+        %% 4. Parse modality
+        % Modalities - the BIDS domains of scantypes
+        ModalitiesUnique = unique(bidsPar.BIDS2LegacyFolderConfiguration(2:end, 2));
+        nModalities = length(ModalitiesUnique);
+        xASL_bids_BIDS2Legacy_ParseModality(BIDS, bidsPar, SubjectVisit, iSubjSess, ModalitiesUnique, nModalities, bOverwrite, pathLegacy_SubjectVisit);
+
     end
-    SubjectVisit = [SubjectID VisitString];
-    xASL_adm_CreateDir(pathLegacy_SubjectVisit);
-    
-    
-    %% 4. Parse modality
-    % Modalities - the BIDS domains of scantypes
-    ModalitiesUnique = unique(bidsPar.BIDS2LegacyFolderConfiguration(2:end, 2));
-    nModalities = length(ModalitiesUnique);
-    xASL_bids_BIDS2Legacy_ParseModality(BIDS, bidsPar, SubjectVisit, iSubjSess, ModalitiesUnique, nModalities, bOverwrite, pathLegacy_SubjectVisit);
-        
+
 end
+% Print new line after track progress bar
+fprintf('   \n');
+
+
+% Get directories of current subject
+SubjectDirs = xASL_adm_GetFileList(fullfile(pathLegacy), ['^.+' x.SUBJECT '.+$'], [],[],true);
```

Comment:
no idea, probably historical, lets remove it unless Michael can explain.

Reply:
Please not 😄 `fullfile` does nothing if the path is okay, but it switches `/` and `\` for windows related things.

Reply:
Makes sense, but in theory, where does this pathLegacy comes from. We should make sure that everything is created using fullfile, so that the issue with `/` is not there in the first place. And all manually supplied paths should be converted upon reading - that way, we don't have to deal with it at every use. But that's just a principle thing, lets keep it here to avoid creating a bug.